### PR TITLE
Separate agent-tooling from azure-monitor-exporter

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: agent/azure-monitor-exporter/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+plugins {
+  id("ai.java-conventions")
+  id("ai.sdk-version-file")
```
Comment (trask) [inline agent/azure-monitor-exporter/build.gradle.kts:3]: we don't need for this module

```suggestion
```

File: agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulse.java [hunk 2]
```diff
@@ -46,7 +46,8 @@ public static QuickPulse create(
       Supplier<String> instrumentationKey,
       @Nullable String roleName,
       @Nullable String roleInstance,
-      boolean useNormalizedValueForNonNormalizedCpuPercentage) {
+      boolean useNormalizedValueForNonNormalizedCpuPercentage,
+      @Nullable String sdkVersion) {
```
Comment (trask) [inline agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulse.java:50]: do we ever pass `null` here? (wondering if we need the `@Nullable`)

Comment (heyams) [reply agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulse.java]: I passed null to [LiveMetricsSpanProcessor]([LiveMetricsSpanProcessor.java](https://github.com/microsoft/ApplicationInsights-Java/pull/2296/commits/3ba0168a4df9f7f03636060adeb36a5318424965#diff-6dca72a20c8b72cb57639d61cd18c481777a894721416fdab81be6a6db6bce58)). Updated it to take sdkVersion as an input parameter now.

File: agent/azure-monitor-exporter/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  testImplementation("org.junit.jupiter:junit-jupiter")
+  testImplementation("com.azure:azure-core-test:1.7.10")
+  testImplementation("org.assertj:assertj-core")
+  testImplementation("org.awaitility:awaitility")
+  testImplementation("org.mockito:mockito-core")
+  testImplementation("uk.org.webcompere:system-stubs-jupiter:1.1.0")
+  testImplementation("io.github.hakky54:logcaptor")
+
+  // needed temporarily while hosting azure-monitor-opentelemetry-exporter in this repo
```
Comment (trask) [inline agent/azure-monitor-exporter/build.gradle.kts:54]:
```suggestion
```

File: agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulse.java [hunk 4]
```diff
@@ -84,7 +86,8 @@ private void initialize(
...
```
Comment (trask) [inline agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulse.java:90]:
```suggestion
      String sdkVersion) {
```

File: agent/azure-monitor-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseIntegrationTests.java [hunk 5]
```diff
@@ -90,7 +93,7 @@ public void testPingWithAuthentication() {
   public void testPingRequestBody() throws InterruptedException {
     CountDownLatch pingCountDown = new CountDownLatch(1);
     String expectedRequestBody =
-        "\\{\"Documents\":null,\"InstrumentationKey\":null,\"Metrics\":null,\"InvariantVersion\":1,\"Timestamp\":\"\\\\/Date\\(\\d+\\)\\\\/\",\"Version\":\"\\(unknown\\)\",\"StreamId\":\"qpid123\",\"MachineName\":\"machine1\",\"Instance\":\"instance1\",\"RoleName\":null\\}";
+        "\\{\"Documents\":null,\"InstrumentationKey\":null,\"Metrics\":null,\"InvariantVersion\":1,\"Timestamp\":\"\\\\/Date\\(\\d+\\)\\\\/\",\"Version\":\"testSdkVersion\",\"StreamId\":\"qpid123\",\"MachineName\":\"machine1\",\"Instance\":\"instance1\",\"RoleName\":null\\}";
```
Comment (trask) [inline agent/azure-monitor-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseIntegrationTests.java:96]: 👍

**Review by trask — APPROVED** [review]

**Pull request merged**